scripts/kconfig/kconfig.py

```diff
@@ -1,22 +1,33 @@
 #!/usr/bin/env python3
 # Modified from: https://github.com/ulfalizer/Kconfiglib/blob/master/examples/merge_config.py
 from kconfiglib import Kconfig, Symbol, BOOL, STRING, TRISTATE, TRI_TO_STR
+import io
 import sys
 
 if len(sys.argv) < 5:
     print('usage: {} Kconfig dotconfig autoconf conf1 [conf2 ...]'
           .format(sys.argv[0]))
     sys.exit(1)
 
+
+# Capture warnings from Kconfiglib, which always go to stderr, into a string.
+# This allows us to turn some of them into errors later.
+#
+# This is a bit hacky, but I'm not sure whether it's worth adding a less
+# dead-simple warning scheme to Kconfiglib for this 1% case.
+stderr_write = sys.stderr.write
+sys.stderr = io.StringIO()
```

Comment:
Have you checked if this works on Windows?

Reply:
Tested now. Works fine.

It's a pure-Python change. Python is pretty flexible in the weird things it allows you to do. :)